test/replication.test.js

```diff
@@ -52,6 +52,121 @@ describe('Replication / Change APIs', function() {
     };
   });
 
+  describe('optimization check rectifyChange Vs rectifyAllChanges', function() {
+    beforeEach(function initialData(done) {
+      var data = [{name: 'foo1', lname: 'bar1'}, {name: 'foo2', lname: 'bar2'}];
+      async.auto({
+        createSourceModel: function(callback) {
+          SourceModel.create(data, callback);
+        },
+        replicate: ['createSourceModel', function(callback, results) {
+            SourceModel.replicate(TargetModel, callback);
+          }]
+      }, function(err, results) {
+        done(err);
+      });
+    });
+
+    it('rectifyOnDelete for Delete should call rectifyChange instead of rectifyAllChanges', function(done) {
+      var count = 0;
+      TargetModel.rectifyChange = function() {
+        // based on destroyAll where condition it should call rectifyChange two times
+        if (++count == 2) {
+          TargetModel.find(function(err, data) {
+            expect(data.length).to.equal(0);
+            return done(err);
+          });
+        }
```

Comment:
This test mixes test setup with assertions, which makes it more difficult to understand.

Here is a better solution (a mock-up to illustrate the idea):

```js
var calls = [];
TargetModel.rectifyChange = function() { calls.push('rectifyChange'); };
TargetModel.rectifyAllChanges = function() { calls.push('rectifyAllChanges'); };

// run destroy & replicate

expect(calls).to.eql(['rectifyChange', 'rectifyChange']);
```